sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -434,6 +434,22 @@ class SessionCatalog(
     externalCatalog.getTable(db, table)
   }
 
+  /**
+   * Retrieve the metadata of an existing permanent table/view. If no database is specified,
+   * assume the table/view is in the current database.
+   */
+  @throws[NoSuchDatabaseException]
+  def getTablesByName(names: Seq[TableIdentifier]): Seq[CatalogTable] = {
+    if (names.nonEmpty) {
+      val db = formatDatabaseName(names.head.database.getOrElse(getCurrentDatabase))
```

Comment:
What if names from different databases? for example:
```scala
Seq("db1.table1", "db2.table1")
```

Reply:
I have considered this case. Could this be happened in the reality use cases? But good catch, it should be handled in code. How about throws an exception?